execution_engine/src/core/engine_state/executable_deploy_item.rs

```diff
@@ -320,7 +322,9 @@ impl ExecutableDeployItem {
                 error::Error::Exec(execution::Error::NoSuchMethod(entry_point_name.to_owned()))
             })?;
 
-        if protocol_data.system_contracts().contains(&contract_hash) {
+        let mut system_hashes = system_contract_registry.into_values();
+
+        if system_hashes.any(|value| value == contract_hash) {
```

Comment:
Maybe a little more canonical to not extract the iterator to a separate variable:
```suggestion
        if system_contract_registry
            .values()
            .any(|&value| value == contract_hash)
        {
```

Reply:
Applied suggestion